src/precice/impl/Participant.cpp

```diff
@@ -103,15 +103,17 @@ void Participant::addWriteData(
     const mesh::PtrMesh &mesh)
 {
   checkDuplicatedData(data);
-  _writeDataContexts[data->getID()] = std::make_unique<WriteDataContext>(data, mesh);
+  auto inserted = _writeDataContexts.insert(std::make_pair(data->getID(), WriteDataContext(data, mesh)));
+  PRECICE_ASSERT(inserted.second, "addWriteData failed.");
```

Comment:
This will only fail if `checkDuplicatedData` doesn't work. So you could add this to the assertion message.

Also you can use `map::emplace()` to avoid using `std::make_pair`.

Reply:
Removed assertion and using `emplace` in  69c6da5b425916ddfdb8faa33c93fbedc5f33d21.